backend/routes/image_generation/image_generation_api.js

```diff
@@ -0,0 +1,129 @@
+import axios from "axios";
+
+const baseUrl = "https://stablehorde.net/api/v2";
+const apiKey = process.env.API_KEY;
+
+let userPrompt = "Pixel Knight";
+let pixelSize = 16;
+
+// Header parameters
+const header = {
+  headers: {
+      "accept": "application/json",
+      "apikey": apiKey,
+      "Content-Type": "application/json",
+  }
+}
+
+// Function to get image id
+const getImageID = async() => {
+
+    // Data payload
+  const data = {
+    "prompt": `pixelart${userPrompt}, ${pixelSize}px, side view, gamedev. game asset, pixelsprite, pixel-art, pixel_art, retro_artstyle, colorful, low-res, blocky, pixel art style, 16-bit graphics ### out of frame, duplicate, watermark, signature, text, error, deformed, sloppy, messy, blurry, noisy, highly detailed, ultra textured, photo, realisticlogo`,
+    "params": {
+      "cfg_scale": 7,
+      "seed": "493768514",
+      "sampler_name": "k_euler_a",
+      "height": 512,
+      "width": 512,
+      "post_processing": [],
+      "steps": 10,
+      "tiling": false,
+      "karras": true,
+      "hires_fix": false,
+      "clip_skip": 1,
+      "n": 1
+    },
+    "allow_downgrade": false,
+    "nsfw": false,
+    "censor_nsfw": true,
+    "trusted_workers": true,
+    "models": [
+      "AIO Pixel Art"
+    ],
+    "r2": true,
+    "replacement_filter": true,
+    "shared": false,
+    "slow_workers": false,
+    "dry_run": false
+  }
+
+  try{
+      const response = await axios.post(`${baseUrl}/generate/async`, data, header);
+      const id = response.data.id;
+      console.log("Generation ID:", id);
+      return id; 
+  } catch(e) {
+    console.error(
+      "Error getting generation ID: " + e.response ? e.response.data : e.message);
+  }
+}
+
+// Function to check the status of the generation
+const checkGenerationStatus = async(id) => {
+  try {
+    const response = await axios.get(`${baseUrl}/generate/check/${id}`, header);
+    return response.data;
+  } catch (e) {
+    console.error("Error checking status:" +  e.response ? e.response.data : e.message);
+    return null;
+  }
+}
+
+// Function to get the full status of the generation
+const getFUllGenerationStatus = async(id) => {
+  try{
+    const response = await axios.get(`${baseUrl}/generate/status/${id}`, header);
+    return response.data;
+  } catch (e) {
+    console.error("Error retrieving full status: " + e.response ? e.response.data : e.message);
+    return null;
+  }
+}
+
+// Poll the status until done
+const pollGenerationStatus = async(id) => {
+  const maxRetries = 10;
```

Comment:
An image generation might in some cases take as long as 10 minutes. This max retries is to restrictive. Also the api request should not wait at all. There should be an api exposed to the user that pretty much matches the one you are calling where they need to poll us to check if its done.